From: "Jyri J. Virkki" <jyri@virkki.com>
Date: Fri, 17 Mar 2017 22:23:53 -0700
Subject: (github #6) Add BITS env var support in Makefile.

BITS can be set to 32 or 64 to select 32bit or 64bit build.

If BITS is unset, it defaults to 64 bit (as before).

If BITS is set to the string default, no option is passed to the
compiler, which means the default for the platform will be built.
---
 Makefile | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index cbd79a9..76e0577 100644
--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,8 @@
 # Other build options:
 #
 #   DEBUG=1 make        to build debug instead of optimized
-#   MM=-m32 make        to build 32bit library
+#   BITS=32 make        to build 32bit library
+#   BITS=default make	to build platform default bitness (32 or 64)
 #
 # Other build targets:
 #
@@ -33,8 +34,16 @@ LIB=-lm
 COM=${CC} $(CFLAGS) $(CPPFLAGS) -Wall ${OPT} ${MM} -std=c99 -fPIC -DBLOOM_VERSION=$(BLOOM_VERSION)
 TESTDIR=$(TOP)/misc/test
 
-ifeq ($(MM),)
+ifeq ($(BITS),)
 MM=-m64
+else ifeq ($(BITS),64)
+MM=-m64
+else ifeq ($(BITS),32)
+MM=-m32
+else ifeq ($(BITS),default)
+MM=
+else
+MM=$(BITS)
 endif
 
 #
